### ISSUE TITLE
borg2: `setup.py` does not support OpenBSD's current versions

### ISSUE THREAD

**Issue by dertuxmalwieder** [borgbackup/borg#8553]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yep.

## Is this a BUG / ISSUE report or a QUESTION?

Bug, basically.

#### Your borg version (borg -V).

None (can't).

#### Operating system (distribution) and version.

OpenBSD 7.6.

## Describe the problem you're observing.

It is currently not possible to run `pipx install borgbackup==2.0.0b14` on OpenBSD:

```
src/borg/crypto/low_level.c:16241:37: error: use of undeclared identifier 'EVP_aes_256_ocb'
```

As OpenBSD defaults to LibreSSL, Borg's `setup.py` contains a workaround; alas, that's outdated:

```python
elif is_openbsd:
        # Use openssl (not libressl) because we need AES-OCB via EVP api. Link
        # it statically to avoid conflicting with shared libcrypto from the base
        # OS pulled in via dependencies.
        crypto_ext_lib = {"include_dirs": ["/usr/local/include/eopenssl30"]}
        crypto_extra_objects += ["/usr/local/lib/eopenssl30/libcrypto.a"]
```

OpenBSD [does not have](https://openports.pl/cat/security) OpenSSL 3.0 anymore (currently, 1.1, 3.2 and 3.3 are there), so trying to use this hard-coded path *will* fail.

## Solution (idea)

Make the crypto library path(s) configurable (or, at least, the `eopenssl30` part), so when OpenBSD drops 3.2 and 3.3, the port maintainer(s) only need to change an environment variable.

**Comment by bket** [comment]
@dertuxmalwieder, please have a look at #8557. Does this fit your need?

**Comment by dertuxmalwieder** [comment]
@bket Looks mostly good to me, although I would probably set the default to `eopenssl32` now as that's the widest supported version...?

**Comment by bket** [comment]
@dertuxmalwieder, you are right. However, I set it to `eopenssl30` as this fits the Vagrant box @ThomasWaldmann is using currently, which is running OpenBSD-7.4 and has no port for OpenSSL-3.2.

Nonetheless, default can be changed easily. I leave it up to @ThomasWaldmann to decide.

**Comment by ThomasWaldmann** [comment]
See my comments in the PR.

**Comment by ThomasWaldmann** [comment]
Fixed by #8557.

**Comment by dertuxmalwieder** [comment]
FYI, it's 35 on OpenBSD 7.8. :)
